doc/source/ray-observability/user-guides/configure-logging.md

```diff
@@ -0,0 +1,382 @@
+
+When running a local Ray cluster, this environment variable should be set before starting the local cluster:
+
+```python
+os.environ["RAY_LOG_TO_STDERR"] = "1"
+ray.init()
+```
+
+When starting a local cluster via the CLI or when starting nodes in a multi-node Ray cluster, this environment variable should be set before starting up each node:
+
+```bash
+env RAY_LOG_TO_STDERR=1 ray start
+```
+
+If using the Ray cluster launcher, you would specify this environment variable in the Ray start commands:
+
+```bash
+head_start_ray_commands:
+    - ray stop
+    - env RAY_LOG_TO_STDERR=1 ray start --head --port=6379 --object-manager-port=8076 --autoscaling-config=~/ray_bootstrap_config.yaml
+
+worker_start_ray_commands:
+    - ray stop
+    - env RAY_LOG_TO_STDERR=1 ray start --address=$RAY_HEAD_IP:6379 --object-manager-port=8076
+```
+
+When connecting to the cluster, be sure to set the environment variable before connecting:
+
+```python
+os.environ["RAY_LOG_TO_STDERR"] = "1"
+ray.init(address="auto")
+```
+
+
+
+## Log deduplication
+
+By default, Ray deduplicates logs that appear redundantly across multiple processes. The first instance of each log message is always immediately printed. However, subsequent log messages of the same pattern (ignoring words with numeric components) are buffered for up to five seconds and printed in batch. For example, for the following code snippet:
+
+```python
+import ray
+import random
+
+@ray.remote
+def task():
+    print("Hello there, I am a task", random.random())
+
+ray.get([task.remote() for _ in range(100)])
+```
+
+The output is as follows:
+
+```bash
+2023-03-27 15:08:34,195	INFO worker.py:1603 -- Started a local Ray instance. View the dashboard at http://127.0.0.1:8265 
+(task pid=534172) Hello there, I am a task 0.20583517821231412
+(task pid=534174) Hello there, I am a task 0.17536720316370757 [repeated 99x across cluster] (Ray deduplicates logs by default. Set RAY_DEDUP_LOGS=0 to disable log deduplication)
+```
+
+This feature is especially useful when importing libraries such as `tensorflow` or `numpy`, which may emit many verbose warning messages when imported. You can configure this feature as follows:
+
+1. Set ``RAY_DEDUP_LOGS=0`` to disable this feature entirely.
+2. Set ``RAY_DEDUP_LOGS_AGG_WINDOW_S=<int>`` to change the agggregation window.
+3. Set ``RAY_DEDUP_LOGS_ALLOW_REGEX=<string>`` to specify log messages to never deduplicate.
+4. Set ``RAY_DEDUP_LOGS_SKIP_REGEX=<string>`` to specify log messages to skip printing.
+
+
+
+## Distributed progress bars (tqdm)
+
+When using `tqdm <https://tqdm.github.io>`__ in Ray remote tasks or actors, you may notice that the progress bar output is corrupted. To avoid this problem, you can use the Ray distributed tqdm implementation at ``ray.experimental.tqdm_ray``:
```

Comment:
```suggestion
When using `tqdm <https://tqdm.github.io>`__ in Ray remote Tasks or Actors, you may notice that the progress bar output is corrupted. To avoid this problem, use the Ray distributed tqdm implementation at ``ray.experimental.tqdm_ray``:
```